core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -1378,6 +1377,14 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     // after rebalancing, we should have reset to the committed positions
     assertEquals(10, testConsumer.committed(Set(tp).asJava).get(tp).offset)
     assertEquals(20, testConsumer.committed(Set(tp2).asJava).get(tp2).offset)
+
+    // In both CLASSIC and CONSUMER protocols, interceptors are executed in poll and close.
+    // However, in the CONSUMER protocol, the assignment may be changed outside of a poll, so
+    // we need to poll once to ensure the interceptor is called.
+    if (groupProtocol.toUpperCase == GroupProtocol.CONSUMER.name) {
```

Comment:
Do we actually need to differentiate between the protocols here? Would it also be OK to call `poll()` irrespectively from the protocol?

Reply:
We can call it irrespective of the protocol. This is just to make the minor difference in behavior explicit. But we can simplify if you prefer

Reply:
I fine either way.